data-prepper-core/src/test/java/org/opensearch/dataprepper/pipeline/router/RouterCopyRecordStrategyTests.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.pipeline.router;
+
+import org.opensearch.dataprepper.model.record.Record;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.opensearch.dataprepper.parser.DataFlowComponent;
+import org.opensearch.dataprepper.pipeline.PipelineConnector;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertFalse;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.event.JacksonEvent;
+
+@ExtendWith(MockitoExtension.class)
+public class RouterCopyRecordStrategyTests {
+
+    @Mock
+    private DataFlowComponent<PipelineConnector> pipelineDataFlowComponent;
+    private Collection<Record> recordsIn;
+
+    private static class TestComponent {
+    }
+
+    @BeforeEach
+    void setUp() {
+        recordsIn = IntStream.range(0, 10)
+                .mapToObj(i -> mock(Record.class))
+                .collect(Collectors.toList());
+    }
+
+    private <C> RouterGetRecordStrategy createObjectUnderTest(Collection<DataFlowComponent<C>> dataFlowComponents) {
+        return new RouterCopyRecordStrategy(dataFlowComponents);
+    }
+
```

Comment:
Please add a test that verifies that the strategy passed into the constructor is not null.

```
strategy = null;
assertThrows(NullPointerException.class, this::createObjectUnderTest);
```